src/convert.rs

```diff
@@ -77,7 +77,7 @@ macro_rules! rant_int_conversions {
             Ok($val)
           };
           (isize, $val:expr) => {
-            Ok(isize($val))
+            $val.try_into().map_err(|_| CastError::Overflow)
```

Comment:
This is a signed narrowing, so it can fail either way around:

```suggestion
            let val = $val;
            val.try_into().map_err(|_| if val < 0 {
              CastError::Underflow
            } else {
              CastError::Overflow
            })
```

Technically you don't need the `let` binding as the macro is used here right now, but I like to make these a bit more robust in case the caller passes a more complex expression later on.

(This is a quick edit based on your work figuring out the types here, so `cargo fmt` may adjust it a bit.)

Reply:
Oh good point!
> (This is a quick edit based on your work figuring out the types here, so cargo fmt may adjust it a bit.)

When I run `cargo fmt` locally it changes over 40 files, I assume that's not what I should do :sweat_smile: